dotnet/dotnet-cloud-run-hello-world/Startup.cs

```diff
@@ -0,0 +1,120 @@
+        public IWebHostEnvironment WebEnvironment { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddRazorPages();
+
+            // Populate Cloud Run environment information: service, revision and Cloud Project
+            //
+
+            // Cloud Run Service
+            string service;
+            try
+            {
+                service = Environment.GetEnvironmentVariable("K_SERVICE");
+            }
+            catch (ArgumentNullException)
+            {
+                service = "???";
+            }
+
+            // Cloud Run Revision
+            string revision;
+            try
+            {
+                revision = Environment.GetEnvironmentVariable("K_REVISION");
+            }
+            catch (ArgumentNullException)
+            {
+                revision = "???";
+            }
+            
+            // Cloud Run project
+            string project = string.Empty;
+            bool projectFound = false;
+            if (WebEnvironment.IsDevelopment())
+            {
+                try
+                {
+                    project = Environment.GetEnvironmentVariable("GOOGLE_CLOUD_PROJECT");
+                    projectFound = true;
+                }
+                catch (ArgumentNullException)
+                {
+                    project = "???";
+                }
+            }
+            else 
+            {
+                if (string.IsNullOrEmpty(project) || service != "???")
+                {
+                    var http = new HttpClient();
+                    http.DefaultRequestHeaders.Add("Metadata-Flavor", new[]{ "Google"});
+
+                    try
+                    {
+                        project = http.GetStringAsync("http://metadata.google.internal/computeMetadata/v1/project/project-id").Result;
+                        projectFound = true;
+                    }
+                    catch (Exception)
+                    {
+                        project = "???";
+                    }
+                }
+            }
+
+            var envInfo = new EnvironmentInfo(service, project, revision, projectFound);
+            services.AddSingleton<IEnvironmentInfo>(envInfo);
+        }
+
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+        {
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+            else
+            {
+                // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
+                app.UseHsts();
+            }
+
+            app.UseHttpsRedirection();

```

Comment:
Cloud Run itself redirects to HTTPS, some of these options may be redundant, but good to keep with a comment for portability.

Reply:
Great point, added a comment. Thanks!